src/domain/pricing/bulk-editor/index.tsx

```diff
@@ -0,0 +1,1052 @@
+        )
+      ) {
+        setPriceChanges({})
+        return true
+      } else {
+        return false
+      }
+    }
+
+    return true
+  }
+
+  const onPriceListSelect = (priceListId: string) => {
+    if (checkForChanges()) {
+      setCurrentPriceListId(priceListId)
+    }
+  }
+
+  const onClose = () => {
+    if (checkForChanges()) {
+      closeEditor()
+    }
+  }
+
+  /**
+   * Save changes callback.
+   * Iterate over a price changes object and prepare MA prices for bulk update.
+   */
+  const onSave = () => {
+    const prices: Partial<MoneyAmount>[] = []
+
+    const _priceChanges = { ...priceChanges }
+
+    priceList.prices
+      .map((p) =>
+        pick(p, [
+          "amount",
+          "variant_id",
+          "region_id",
+          "min_quantity",
+          "max_quantity",
+          "currency_code",
+        ])
+      )
+      .map((plPrice) => {
+        if (
+          plPrice.region_id &&
+          plPrice.min_quantity === null &&
+          plPrice.max_quantity === null
+        ) {
+          const priceKey = getPriceKey(plPrice.variant_id, plPrice.region_id)
+          if (priceKey in _priceChanges) {
+            // MA is deleted
+            if (_priceChanges[priceKey] === undefined) {
+              return
+            }
+
+            // MA is updated
+
+            let preparedAmount = Math.round(
+              parseFloat(_priceChanges[priceKey]) *
+                10 **
+                  currencies[plPrice.currency_code.toUpperCase()].decimal_digits
+            )
+
+            if (isNaN(preparedAmount)) {
+              return
+            }
+
+            prices.push({
+              variant_id: plPrice.variant_id,
+              region_id: plPrice.region_id,
+              amount: preparedAmount,
+            })
+
+            delete _priceChanges[priceKey]
+          } else {
+            prices.push(plPrice)
+          }
+        } else {
+          prices.push(plPrice)
+        }
+      })
+
+    // entries that are left --> MAs to be created
+    Object.entries(_priceChanges).map(([k, amount]) => {
+      const [variantId, regionId] = k.split("-")
+
+      const reg = regions?.find((r) => r.id === regionId)!
+      const curr = currencies[reg.currency_code.toUpperCase()]
+
+      const preparedAmount = Math.round(
+        parseFloat(amount) *
+          10 ** currencies[curr.code.toUpperCase()].decimal_digits
+      )
+
+      if (isNaN(preparedAmount)) {
+        return
+      }
+
+      prices.push({
+        variant_id: variantId,
+        region_id: regionId,
+        amount: preparedAmount,
+      })
+    })
+
+    prices.forEach((p) => {
+      delete p.id
+      if (p.region_id) {
+        delete p.currency_code
+      }
+    }) // new records will be created
+
+    updatePriceList.mutate(
+      { prices, override: true },
+      { onSuccess: () => closeEditor() }
+    )
+  }
+
+  const displayRegions = activeRegions
```

Comment:
r, i, a, b represents all an activeRegion but are named differently, goes with the one letter comments.

Reply:
Naming is shard :D ... I agree 100% that we should avoid one letter naming in most cases but sometimes it is OK for one liner arrow functions to be an exception to the rule.

Reply:
We can agree on one level array function? because otherwise we end up with things like a.id === r or r.id and in that case for example I have to go back into the code itself to see what does that means 😂

And even on one level we would have to go back to the code and implementation to see what to means since out of context I have no clue. Typic example

Reply:
Si lets settle on agree for one liner arrow function 💪